data_steward/utils/bq.py

```diff
@@ -286,3 +286,64 @@ def get_latest_validation_dataset_id(project_id):
             return sorted(validation_datasets, key=lambda x: x[0],
                           reverse=True)[0][1]
     return None
+
+
+def create_dataset(project_id,
+                   dataset_id,
+                   description=None,
+                   friendly_name=None,
+                   overwrite_existing=None):
+    """
+    Creates a new dataset
+
+    :param project_id: name of the project to create dataset in, defaults to the currently configured project if missing.
+    :param dataset_id: name to give the new dataset - required
+    :param description: dataset description - required
+    :param friendly_name: user friendly name for dataset - optional
+    :param overwrite_existing: determine if dataset should be overwritten if already exists, defaults to true/overwrite
+    :return: a new dataset returned from the API
+    """
+    if dataset_id is None:
+        raise RuntimeError("Cannot create a dataset without a name")
+
+    if description is None:
+        raise RuntimeError("Will not create a dataset without a description")
+
+    if project_id is None:
+        LOGGER.info(f"You should specify project_id for a reliable experience."
+                    f"Defaulting to {os.environ.get(PROJECT_ID)}.")
+        project_id = os.environ.get(PROJECT_ID)
```

Comment:
It looks like a lot of these checks are already in the `define_dataset` function. It might be better to call that function within this to be consistent and reuse code.

Reply:
This is a good idea, I will make that change!

Reply:
Quick question-- `define_dataset` assumes the dataset does not exist and raises a conflict if it already exists. `create_dataset` was initially written in `bq_utils` to default to overwrite and does not throw a conflict if the dataset exists (which is what I maintained in the function in `bq`. Do you think this could cause issues or should it default the same and throw a conflict if the dataset exists?

Reply:
One more thing- `define_dataset` requires a lable_or_tag, which `create_dataset` previously did not. And neither does `create_sandbox`, let me know if they should all require, which will need to adjust the usage as well.

Reply:
I think it would be good to have a check if it exists before starting to create the dataset. If `overwrite_existing` is set to `False` and it exists, an error should be raised stating that the dataset already exists. Is this the same error raised by `define_dataset`? 
It's good to add the `label_or_tag` parameter as well since we want to enforce this practice (sandbox should have the `sandbox` tag I think). Try making the changes incl. usages, but if the scope blows up a lot, add it as a new ticket and a different PR.

Reply:
They are the same errors raised if `overwrite_existing` is set to False and it already exists in `define_dataset`, my only concern is if someone wants to set `overwrite_existing` to `True` (not sure if that will even be wanted). Then the function `create_dataset` would not work with `define_dataset`. But for now, all `sandbox.create_sandbox_dataset` sets that parameter to `False` so that won't be an issue with any current usages.

There is no sandbox tag, but I can define that in `create_sandbox_dataset`. Is there anything specific I should set these to? For now I am leaving as blank strings. 

Take a look when you get the chance and let me know if my code makes more sense now using `define_dataset`.

Reply:
I think there can be a check for existence before that step. `define_dataset` should not need to be called if the dataset already exists.
I think you can leave the labels/tags blank for now, at least until we come up with a convention.

Reply:
I've added in a check!